src/main/kotlin/com/cognifide/gradle/aem/AemTaskFacade.kt

```diff
@@ -14,41 +15,61 @@ import com.cognifide.gradle.aem.tooling.rcp.Rcp
 import com.cognifide.gradle.aem.tooling.sync.Sync
 import com.cognifide.gradle.aem.tooling.vlt.Vlt
 import com.fasterxml.jackson.annotation.JsonIgnore
-import java.io.Serializable
 import org.gradle.api.Project
 import org.gradle.api.Task
 import org.gradle.api.UnknownTaskException
 import org.gradle.api.plugins.JavaPlugin
 import org.gradle.api.tasks.TaskProvider
-import org.gradle.api.tasks.bundling.Jar
-import org.gradle.api.tasks.testing.Test
+import java.io.Serializable
 
 @Suppress("TooManyFunctions")
-class AemTaskFacade(private val aem: AemExtension) : Serializable {
+class AemTaskFacade(val aem: AemExtension) : Serializable {
 
     @JsonIgnore
     val project = aem.project
 
-    private val bundleMap = mutableMapOf<String, BundleJar>()
+    // Bundle plugin shorthands
 
-    val bundles: List<BundleJar>
-        get() = getAll(Jar::class.java).map { bundle(it) }
+    val bundles get() = getAll(BundleCompose::class.java)
 
-    internal fun jarsAsBundles() {
-        initializeJarsAsBundles()
-        project.afterEvaluate {
-            finalizeJarsAsBundles()
-        }
+    fun bundleCompose(configurer: BundleCompose.() -> Unit) = named(BundleCompose.NAME, configurer)
+
+    fun bundleExportEmbed(dependencyNotation: String, vararg pkgs: String) {
```

Comment:
this embedding move to here because previously single task was building bundle, jar in the end runs bnd tool
now it is done separately so that adding project dependency was too late here as of we have task configuration avoidance, so that that nice signature embed package need to be moved outside task scope (as of project.dependencies.add is task agnostic)